src/sinks/amqp/config.rs

```diff
@@ -0,0 +1,102 @@
+use crate::{
+    amqp::AMQPConfig,
+    codecs::EncodingConfig,
+    config::{DataType, GenerateConfig, Input, SinkConfig, SinkContext},
+    sinks::{Healthcheck, VectorSink},
+};
+use codecs::TextSerializerConfig;
+use futures::FutureExt;
+use std::sync::Arc;
+use vector_config::configurable_component;
+use vector_core::config::AcknowledgementsConfig;
+
+use super::sink::AMQPSink;
+
+/// Configuration for the `amqp` sink.
+///
+/// Supports AMQP version 0.9.1
+#[configurable_component(sink)]
+#[derive(Clone, Debug)]
+pub struct AMQPSinkConfig {
+    /// The exchange to publish messages to.
+    pub(crate) exchange: String,
+
+    /// Template use to generate a routing key which corresponds to a queue binding.
+    // TODO: We will eventually be able to add metadata on a per-field basis, such that we can add metadata for marking
+    // this field as being capable of using Vector's templating syntax.
```

Comment:
Is there an issue/epic we can reference in the comment ?

Reply:
@spencergilbert Do you know?

Reply:
Commented below with a suggested change (or just the annotation), and a question regarding the field.